File: Masjidly - Official Masjid Prayer Times/Domain/MonthName.swift
import Foundation

enum MonthName: String, CaseIterable, Sendable {
    case january, february, march, april, may, june
    case july, august, september, october, november, december

    static func from(monthNumber: Int) -> MonthName? {
        switch monthNumber {
        case 1: return .january
        case 2: return .february
        case 3: return .march
        case 4: return .april
        case 5: return .may
        case 6: return .june
        case 7: return .july
        case 8: return .august
        case 9: return .september
        case 10: return .october
        case 11: return .november
        case 12: return .december
        default: return nil
        }
    }
}

File: apps/android/app/src/main/kotlin/com/mikhailspeaks/masjidly/domain/MonthName.kt
package com.mikhailspeaks.masjidly.domain

enum class MonthName(val rawValue: String) {
    JANUARY("january"),
    FEBRUARY("february"),
    MARCH("march"),
    APRIL("april"),
    MAY("may"),
    JUNE("june"),
    JULY("july"),
    AUGUST("august"),
    SEPTEMBER("september"),
    OCTOBER("october"),
    NOVEMBER("november"),
    DECEMBER("december"),
    ;

    companion object {
        fun from(monthNumber: Int): MonthName? = when (monthNumber) {
            1 -> JANUARY
            2 -> FEBRUARY
            3 -> MARCH
            4 -> APRIL
            5 -> MAY
            6 -> JUNE
            7 -> JULY
            8 -> AUGUST
            9 -> SEPTEMBER
            10 -> OCTOBER
            11 -> NOVEMBER
            12 -> DECEMBER
            else -> null
        }
    }
}
